src/create-matcher.js

```diff
@@ -52,7 +52,8 @@ export function createMatcher (routes: Array<RouteConfig>): Matcher {
     location: Location
   ): Route {
     const { query, hash, params } = location
-    const { redirect } = record
+    const { redirect: originalRedirect } = record
+    const redirect = typeof originalRedirect === 'function' ? originalRedirect() : originalRedirect
```

Comment:
@yyx990803 Sure it would be useful, but the target route is `match()`-ed after consuming the redirect option... and `this` context is not available here, so I don't know what can be passed...